instrumentation/play-ws/play-ws-1.0/javaagent/src/test/groovy/PlayWsClientTest.groovy

```diff
@@ -46,18 +46,17 @@ class PlayJavaStreamedWsClientTest extends PlayWsClientTestBase {
   StandaloneWSClient wsClient
 
   @Override
-  int doRequest(String method, URI uri, Map<String, String> headers, Closure callback) {
+  int doRequest(String method, URI uri, Map<String, String> headers = [:], Consumer<Integer> callback = null) {
     StandaloneWSRequest wsRequest = wsClient.url(uri.toURL().toString()).setFollowRedirects(true)
 
     headers.entrySet().each { entry -> wsRequest.addHeader(entry.getKey(), entry.getValue()) }
     StandaloneWSResponse wsResponse = wsRequest.setMethod(method).stream()
-      .whenComplete({ response, throwable ->
-        callback?.call()
-      }).toCompletableFuture().get(5, TimeUnit.SECONDS)
+      .toCompletableFuture().get(5, TimeUnit.SECONDS)
 
     // The status can be ready before the body so explicitly call wait for body to be ready
     wsResponse.getBodyAsSource().runFold("", { acc, out -> "" }, materializer)
       .toCompletableFuture().get(5, TimeUnit.SECONDS)
+    callback?.accept(wsResponse.getStatus())
```

Comment:
While the play tests seem to have regressed in their async-ness, presumably they're already sort of broken since I'd expect this "body to be ready" caveat to be required for both sync and async. Will address when splitting out sync / async handlers.